apps/verticalhome/test/marionette/group_collapse_icon_launch_test.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+var assert = require('assert');
+
+marionette('Vertical - Group', function() {
+
+  var client = marionette.client(require(__dirname + '/client_options.js'));
+  var actions, home, system;
+
+  setup(function() {
+    actions = client.loader.getActions();
+    home = client.loader.getAppClass('verticalhome');
+    system = client.loader.getAppClass('system');
+    system.waitForStartup();
+
+    client.apps.launch(home.URL);
+    home.waitForLaunch();
+  });
+
+  test('collapse the first group', function() {
+    // Get a reference for the first icon
+    var icon = client.helper.waitForElement(home.Selectors.firstIcon);
+
+    // Retrieve the icon's manifest URL
+    var manifestURL = icon.getAttribute('data-identifier');
+    manifestURL = manifestURL.slice(0, manifestURL.lastIndexOf('-'));
+
+    // Collapse first group
+    var collapse = client.findElement(home.Selectors.groupToggle);
+    actions.tap(collapse).wait(1).perform();
+
+    // Tap the icon
+    actions.tap(icon).perform();
+
+    // Verify the app launched
+    client.waitFor(function() {
+      // go to the system app
+      client.switchToFrame();
+
+      var frames = client.findElements('iframe');
```

Comment:
For some reason if I just used findElement('iframe[mozapp="' + manifestURL + '"]'), I don't get a match. Any suggestions welcome, otherwise this works fine.